resources/osl_hookshot.rb

```diff
@@ -0,0 +1,157 @@
+# osl_hookshot
+#
+# Hookshot is a webhook bot for a Matrix server.
+# Designed to only connect to a Synapse instance
+
+resource_name :osl_hookshot
+provides :osl_hookshot
+unified_mode true
+
+default_action :create
+
+property :config, Hash, required: true
+property :container_name, String, name_property: true
+property :port, Integer, default: 9993
+property :port_webhook, Integer, default: 9000
+property :port_metric, Integer, default: 9001
+property :port_widget, Integer, default: 9002
+property :port_host, Integer, default: 8008
+property :host_domain, String, required: true
+property :host_name, String, default: lazy { "matrix-synapse-#{host_domain}" }
+property :host_path, String, default: lazy { "/opt/synapse-#{host_name}" }
+property :host_network, String, default: lazy { "synapse-network-#{host_name}" }
+property :key_appservice, String, default: lazy { osl_matrix_genkey(host_name + container_name) }
+property :key_homeserver, String, default: lazy { osl_matrix_genkey(host_network + container_name) }
+
+action :create do
+  # Pull down the Hookshot image
+  docker_image 'halfshot/matrix-hookshot'
+
+  # Parse out all of the instance of userIdPrefix in order to modify the appservice registration
+  arrAppServiceUsers = []
+  # Loop over the root hash
+  new_resource.config.each_value do |val|
+    # Check to see if the sub-hash has a key called userIDPrefix
+    unless val.is_a?(Hash) && val.key?('userIdPrefix')
+      next
+    end
+    # Add the userIDPrefix
+    arrAppServiceUsers.push({
+      exclusive: true,
+      regex: "\'@#{val['userIdPrefix']}_.*\'",
+    })
+  end
+
+  # Add in something if there was nothing given
+  unless arrAppServiceUsers
+    arrAppServiceUsers.push({
+      exclusive: true,
+      regex: '\'@webhook_.*\'',
+    })
+  end
+
+  # Generate the app service file
+  osl_synapse_appservice(
+    'hookshot',
+    "http://#{new_resource.container_name}:#{new_resource.port}",
+    new_resource.key_appservice,
+    new_resource.key_homeserver,
+    'hookshot',
+    {
+      users: arrAppServiceUsers,
+    }
+  )
+
+  # Auto configuration for the homeserver, ignore if already set
+  # Passkey settings
+  # Not usually needed to be modified by user, as we generate our own, but just in case.
+  new_resource.config.merge!({
```

Comment:
You shouldn't be directly modifying the parameters in `new_resource` Can you please make a local variable to do this with?